oauth2/oauth2-jose/src/main/java/org/springframework/security/oauth2/jwt/JwtDecoders.java

```diff
@@ -15,26 +15,33 @@
  */
 package org.springframework.security.oauth2.jwt;
 
+import static org.springframework.security.oauth2.jwt.NimbusJwtDecoder.withJwkSetUri;
+
 import java.net.URI;
 import java.util.Map;
 
 import org.springframework.core.ParameterizedTypeReference;
+import org.springframework.http.HttpStatus;
 import org.springframework.http.RequestEntity;
 import org.springframework.security.oauth2.core.OAuth2TokenValidator;
+import org.springframework.web.client.HttpClientErrorException;
 import org.springframework.web.client.RestTemplate;
 import org.springframework.web.util.UriComponentsBuilder;
 
-import static org.springframework.security.oauth2.jwt.NimbusJwtDecoder.withJwkSetUri;
-
 /**
  * Allows creating a {@link JwtDecoder} from an
  * <a href="https://openid.net/specs/openid-connect-discovery-1_0.html#ProviderConfig">OpenID Provider Configuration</a>.
  *
  * @author Josh Cummings
+ * @author Rafiullah Hamedy
```

Comment:
There is also `ReactiveJwtDecoders` to take care of. 

In that case, you might consider moving `getOpenIdConfiguration` and `getOAuth2Configuration` to a package-private class, since `JwtDecoders` and `ReactiveJwtDecoders` are in the same package, though abstracting this out is something that I think can wait for another PR.

Reply:
After looking at the `ReactiveJwtDecoders`, I agree it does make sense to refactor and reuse. I would be happy to work on a subsequent PR for it whenever we decide to create the issue for it 👍